### PR TITLE
Added missing destructor-functions for memory-based types

### PR DESCRIPTION
**Pull request by Tikrass**
In the Compiler I added destructor methods `delete()` which call the destructor of the underlying memory-allocation or types recursively. Namely, I changed `Matrix`, `MultiArray`, `VectorArray`, `sfloatArray`, `sfloatMatrix`, `sfixArray`, `sfixMatrix`, `MemFloat`, `MemFix`.

### CODE REVIEW

File: Compiler/types.py [hunk 1]
```diff
@@ -2354,6 +2354,10 @@ def __init__(self, rows, columns, value_type, address=None):
             value_type = _types[value_type]
         self.value_type = value_type
         self.address = Array(rows * columns, value_type, address).address
+        
+    def delete(self):
```
Comment (mkskeller) [inline Compiler/types.py]: Since this uses Array internally, I would have it call the respective function there.

File: Compiler/types.py [hunk 2]
```diff
@@ -2436,13 +2449,17 @@ def __init__(self, rows, columns):
...
+        self.multi_array.delete()
 
     def __getitem__(self, index):
         return sfloatArray(self.columns, self.multi_array[index].address)
 
 class sfixArray(Array):
     def __init__(self, length, address=None):
         self.array = Array(length, sint, address)
+        self.address = self.array.address
```
Comment (mkskeller) [inline Compiler/types.py]: This seems redundant.

Comment (Tikrass) [reply Compiler/types.py]: Either we make `address` available to `self` such that the inherited `Array.delete()` function works or we call delete on `self.array`. Now that you mention it, last looks better.

File: Compiler/types.py (hunk 1 above)
Comment (Tikrass) [reply Compiler/types.py]: `Matrix` does not inherit from `Array`. I am not sure, what is the best practice here.

**Comment by Tikrass** [comment]
@mkskeller In `Matrix` I call the respective destructor of array. If you agree on these changes you may merge them.

**Comment by Tikrass** [comment]
Are there any plans to merge this request? Otherwise I would close it, because it is somehow blocking my master branch. 
Regards

**Pull request closed without merging**